pallets/identity/src/lib.rs

```diff
@@ -1648,11 +1659,40 @@ impl<T: Trait> Module<T> {
         expiry: Option<T::Moment>,
     ) -> DispatchResult {
         Self::ensure_authorized_cdd_provider(issuer)?;
+        // Ensure cdd_id uniqueness for a given target DID.
+        Self::ensure_cdd_id_uniqueness(&claim, target)?;
 
         Self::base_add_claim(target, claim, issuer, expiry);
         Ok(())
     }
 
+    /// Enforce CDD_ID uniqueness for a given target DID.
+    ///
+    /// # Errors
+    /// - `CDDIdNotUniqueForIdentity` is returned when new cdd claim's cdd_id
+    /// doesn't match the existing cdd claim's cdd_id.
+    fn ensure_cdd_id_uniqueness(claim: &Claim, target: IdentityId) -> DispatchResult {
+        if let Claim::CustomerDueDiligence(cdd_id) = claim {
+            ensure!(
+                !Self::base_fetch_valid_cdd_claims(target, 0.into(), None).any(|id_claim| {
+                    if let Claim::CustomerDueDiligence(c_id) = id_claim.claim {
+                        // Allowing default cdd_id.
+                        if !c_id.is_default_cdd()
+                            && !SYSTEMATIC_ISSUERS
+                                .iter()
+                                .any(|si| si.as_id() == id_claim.claim_issuer)
+                        {
+                            return c_id != *cdd_id;
+                        }
+                    }
+                    false
+                }),
+                Error::<T>::CDDIdNotUniqueForIdentity
+            );
```

Comment:
```suggestion
            ensure!(
                cdd_id.is_default_cdd()
                    || Self::base_fetch_valid_cdd_claims(target, 0.into(), None).all(|id_claim| {
                        if let Claim::CustomerDueDiligence(c_id) = id_claim.claim {
                            return c_id.is_default_cdd() || c_id == *cdd_id;
                        }
                        true
                    }),
                Error::<T>::CDDIdNotUniqueForIdentity
            );
```

Reply:
I don't see any need to modify it.

Reply:
This simplifies the code...
The logic goes from 13 LOC to 7 and from double negation to normal.